src/coverlet.core/Symbols/CecilSymbolHelper.cs

```diff
@@ -202,6 +202,89 @@ Single branch
+			                    IL_002b: stfld object ...::'<>s__1'
+			                    // <>s__2 = 1;
+			                    IL_0030: ldarg.0
+			                    IL_0031: ldc.i4.1
+			                    IL_0032: stfld int32 ...::'<>s__2'      <- store 1 into <>s__2
+			                    // (no C# code)
+			                    IL_0037: leave.s IL_0039
+		                    } // end handle
+
+                            // int num2 = <>s__2;
+                            IL_0039: ldarg.0
+		                    IL_003a: ldfld int32 ...::'<>s__2'          <- load <>s__2 value and check if 1
+		                    IL_003f: stloc.3
+                            // if (num2 == 1)
+		                    IL_0040: ldloc.3
+		                    IL_0041: ldc.i4.1
+		                    IL_0042: beq.s IL_0049                      <- if <>s__2 value is 1 go to exception handler code
+
+                            IL_0044: br IL_00d6
+
+                            IL_0049: nop                                <- start exception handler code
+                            ...
+
+                            
+                            So starting from branch instruction 'beq.s', we can go back to starting block instruction 
+                            which is always 5 step before and then check if this istruction is the end of an exception handler block
+                        */
+
+                        int branchIndex = instructions.IndexOf(instruction);
+                        if (
+                                branchIndex >= 5 &&
+                                instructions[branchIndex - 5].OpCode == OpCodes.Ldarg &&
+                                methodDefinition.Body.ExceptionHandlers.Any(h => h.HandlerEnd == instructions[branchIndex - 5])
+                            )
+                        {
+                            continue;
+                        }
+
+                        /* 
+                            In case of exception re-thrown inside the catch block, 
+                            the compiler generates a branch to check if the exception reference is null.
+                            
+                            A sample of generated code:
+                         
+                            IL_00b4: isinst [System.Runtime]System.Exception
+                            IL_00b9: stloc.s 6
+                            // if (ex == null)
+                            IL_00bb: ldloc.s 6
+                            // (no C# code)
+                            IL_00bd: brtrue.s IL_00c6
+
+                            So we can go back to previous instructions and skip this branch if recognize that type of code block
+                        */
+
+                        if (
+                                branchIndex >= 3 && // avoid out of range exception (need almost 3 instruction before the branch)
+                                instructions[branchIndex - 3].OpCode == OpCodes.Isinst &&
```

Comment:
I'd add bounds also here if for some reason we fail to get index we go on with no problem, or are you sure about every possible case?